backend/src/main/java/io/github/projektalmanac/amoxtli/backend/service/UserService.java

```diff
@@ -305,5 +311,89 @@ public SessionTokenDto iniciarSesion(CredencialesDto credenciales){ //hash passw
 
 
     }
+    /**/
+    public ValidaPuedeIntercambiar200ResponseDto validaIntercambio(Integer id) {
+
+        Optional<User> userOpt = userRepository.findById(id);
+        if (userOpt.isPresent()) {
+            throw new UserNotFoundException(id);
+        }
```

Comment:
```suggestion
        if (userOpt.isEmpty()) {
            throw new UserNotFoundException(id);
        }
```

Estás lanzando `UserNotFoundException` si el repositorio te regresa un usuario no null